wrappers/azurekeyvault/azurekeyvault.go

```diff
@@ -118,6 +123,8 @@ func (v *Wrapper) SetConfig(config map[string]string) (map[string]string, error)
 	}
 
 	switch {
+	case v.keyNotRequired:
```

Comment:
Can there be a case where a key is not required but provided anyways such that it's supposed to be used? If so, we might want to move this towards the bottom of the switch, right above the default case.

Reply:
Yep, I could see there being a case for that. I've moved it as you've suggested in a34f78c.